04/app.js

```diff
@@ -1,5 +1,70 @@
-document.addEventListener('DOMContentLoaded', init);
+document.addEventListener('DOMContentLoaded', function() {
+    const form = document.querySelector('.form');
+    const apiKey = 'f244bf8743ea42788e8bdc3e9aa9c35d'; 
 
-function init() {
-    console.log('DOM');
-}
\ No newline at end of file
+    form.addEventListener('submit', function(event) {
+        event.preventDefault();
+
+        const latitude = document.querySelector('.form__field--lat').value;
+        const longitude = document.querySelector('.form__field--lng').value;
+        const weatherUrl = `https://api.weatherbit.io/v2.0/current?key=${apiKey}&lat=${latitude}&lon=${longitude}&units=M&lang=pl`;
+
+        // Rozpislem komentarze ponizej, zeby utrwalic sobie kolejnosc wykonywanych operacji poniewaz jest to dosc skomplikowane
+        // I bylo duzo bledow podczas pisania tego kodu :D
+        // Wysyłamy zapytanie do API pogodowego
+        fetch(weatherUrl)
+            .then(response => {
+                // Sprawdzamy czy odpowiedz jest pomyslna
+                if (!response.ok) {
+                    // Jesli odpowiedz nie jest pomyslna, wyrzucamy błąd wraz z odpowiedza
+                    throw new Error('Network response was not ok ' + response.statusText);
+                }
+                // Jesli odpowiedz jest ok, zwracamy dane w formacie JSON
+                return response.json();
+            })
+            .then(data => {
+                // Sprawdzamy czy w odpowiedzi znajduje sie tablica danych i czy zawiera przynajmniej jeden element
+                if (data.data && data.data.length > 0) {
+                    // Jesli dane sa prawidlowe, wywoluje funkcje do wyswietlenia informacji o pogodzie
+                    displayWeather(data);
+                    // Pokazujemy sekcje z informacjami o pogodzie
+                    document.querySelector('.weather').style.display = 'block';
```

Comment:
Sporo masz funkcję `displayWeather` to uważam, że wew. tej funkcji powinno być pokazywanie tego elementu :)